gui/src/components/onboarding/pages/body-proportions/scaled-steps/ManualHeightStep.tsx

```diff
@@ -104,17 +103,23 @@ export function ManualHeightStep({
                 'onboarding-scaled_proportions-manual_height-height-v2'
               )}
               valueLabelFormat={(value) =>
-                isNaN(value)
-                  ? l10n.getString(
-                      'onboarding-scaled_proportions-manual_height-unknown'
-                    )
-                  : mFormat.format(value)
+                isNaN(value) ? (
+                  l10n.getString(
+                    'onboarding-scaled_proportions-manual_height-unknown'
+                  )
+                ) : (
+                  <HeightDisplay
+                    height={value}
+                    heightUnit={currentUnit}
+                    unitDisplay="narrow"
+                  />
+                )
               }
-              min={MIN_HEIGHT}
-              max={4}
-              step={0.01}
-              showButtonWithNumber
-              doubleStep={0.1}
+              min={convert(MIN_HEIGHT, 'm').to(currentUnit)}
+              max={convert(4, 'm').to(currentUnit)}
+              step={currentUnit === 'm' ? 0.01 : 1}
+              showButtonWithNumber={currentUnit === 'm' ? 10 : 12}
+              doubleStep={currentUnit === 'm' ? 0.1 : 12}
```

Comment:
All this conversion feels super jank ngl, I feel like we should just keep the internal representation as meters and convert to feet and inches for the display. We could probably switch the increment and do some rounding to imperial units tho, I'm really not sure what's best.